R/Task_operators.R

```diff
@@ -1,40 +1,43 @@
-#' Get a summarizing task description.
+#' @title Get a summarizing task description.
+#'
+#' @description For ease of usage you can also pass in a task description which will
+#' be returned unchanged.
 #'
 #' @template arg_task_or_desc
-#' @return [\code{\link{TaskDesc}}].
+#' @return ret_taskdesc
 #' @export
 #' @family task
 getTaskDescription = function(x) {
-  UseMethod("getTaskDescription")
-}
-
-#' @export
-getTaskDescription.default = function(x) {
-  x$task.desc
-}
-
-#' @export
-getTaskDescription.TaskDesc = function(x) {
-  x
+  checkTask(x, allow.desc = TRUE)
```

Comment:
You are dodging S3 here? Should be `getTaskDescription.Task()` and `getTaskDescription.TaskDescription()`.